Source/WebCore/platform/image-decoders/jpegxl/JxlDecoderPtr.h

```diff
@@ -28,65 +28,24 @@
 #if USE(JPEGXL)
 
 #include <algorithm>
+#include <memory>
 
 #if PLATFORM(COCOA)
 #import <pal/spi/cocoa/AppleJPEGXL/AppleJPEGXLSPI.h>
 #else
 #include <jxl/decode.h>
 #endif
 
-class JxlDecoderPtr {
+class JXLDecoderDeleter {
 public:
-    JxlDecoderPtr() = default;
-
-    JxlDecoderPtr(JxlDecoder* decoder)
-        : m_decoder(decoder)
-    {
-    }
-
-    JxlDecoderPtr(const JxlDecoderPtr&) = delete;
-
-    JxlDecoderPtr(JxlDecoderPtr&& other)
-        : m_decoder(other.m_decoder)
-    {
-        other.m_decoder = nullptr;
-    }
-
-    JxlDecoderPtr& operator=(const JxlDecoderPtr&) = delete;
-
-    JxlDecoderPtr& operator=(JxlDecoderPtr&& other)
-    {
-        reset();
-        std::swap(m_decoder, other.m_decoder);
-        return *this;
-    }
-
-    ~JxlDecoderPtr()
+    void operator()(JxlDecoder* decoder)
     {
-        reset();
+        JxlDecoderDestroy(decoder);
     }
-
-    void reset()
-    {
-        if (m_decoder)
-            JxlDecoderDestroy(m_decoder);
-        m_decoder = nullptr;
-    }
-
-    operator bool() const
-    {
-        return m_decoder;
-    }
-
-    JxlDecoder* get() const
-    {
-        return m_decoder;
-    }
-
-private:
-    JxlDecoder* m_decoder { nullptr };
 };
 
+using JxlDecoderPtr = std::unique_ptr<JxlDecoder, JXLDecoderDeleter>;
```

Comment:
Could we get away with using `jxl/decode_cxx.h` directly?

Reply:
Nope :)
1. We have a policy of not using C++ APIs from other libraries/frameworks, because of C++'s binary compatibility problems
2. AppleJPEGXL doesn't even include that header in the first place

Reply:
TIL for both, makes sense!